### PR TITLE
fix(ci): block workflows on forks

### LINKED ISSUE #2586: [meta] CI runs on forks
We should be ignoring forks for the following types of actions:

- https://github.com/bsmth/interactive-examples/actions/workflows/stage-deploy.yml
- https://github.com/bsmth/interactive-examples/actions/workflows/prod-deploy.yml
- CodeQL?

### PR DESCRIPTION
**Pull request by StyleShit**
Closes #2586

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -24,6 +24,7 @@ jobs:
   analyze:
     name: Analyze
     runs-on: ubuntu-latest
+    if: startsWith( github.repository, 'mdn/' )
```
Comment (caugner) [inline .github/workflows/codeql-analysis.yml]: Let's be consistent with [workflows in other repos](https://github.com/mdn/content/blob/6fa75c683af3831c0e988a9270e3b870852dc9a1/.github/workflows/markdown-lint-fix.yml#L10):

```suggestion
    if: github.repository == 'mdn/interactive-examples'
```

File: .github/workflows/prod-deploy.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ jobs:
       id-token: write
 
     runs-on: ubuntu-latest
-    if: github.triggering_actor != 'dependabot[bot]'
+    if: github.triggering_actor != 'dependabot[bot]' && startsWith( github.repository, 'mdn/' )
```
Comment (caugner) [inline .github/workflows/prod-deploy.yml]: dito

File: .github/workflows/stage-deploy.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ jobs:
       contents: read
       id-token: write
     runs-on: ubuntu-latest
-    if: github.triggering_actor != 'dependabot[bot]'
+    if: github.triggering_actor != 'dependabot[bot]' && startsWith( github.repository, 'mdn/' )
```
Comment (caugner) [inline .github/workflows/stage-deploy.yml]: dito

**Review by caugner — CHANGES_REQUESTED** [review]
Thank you! 👍 

LGTM, but let's use the same condition as in other repos, to be consistent.

I can see the advantage of `startsWith`, but we don't have plans to rename the repos.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (StyleShit) [reply .github/workflows/codeql-analysis.yml]: sure, done :)

File: .github/workflows/prod-deploy.yml (hunk 2 above)
Comment (StyleShit) [reply .github/workflows/prod-deploy.yml]: Done

File: .github/workflows/stage-deploy.yml (hunk 3 above)
Comment (StyleShit) [reply .github/workflows/stage-deploy.yml]: Done

**Review by caugner — APPROVED** [review]

**Comment by bsmth** [comment]
Thanks, both! 👍🏻

**Pull request merged**